rust/datafusion/tests/sql.rs

```diff
@@ -508,6 +508,26 @@ async fn csv_query_avg_multi_batch() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn csv_query_nullif_divide_by_0() -> Result<()> {
+    let mut ctx = ExecutionContext::new();
+    register_aggregate_csv(&mut ctx)?;
+    let sql = "SELECT c8/nullif(c7, 0) FROM aggregate_test_100";
+    let actual: Vec<_> = execute(&mut ctx, sql)
+        .await
+        .iter()
+        .map(|x| x[0].clone())
+        .collect();
+    let actual = actual.join("\n");
+    let expected = "1722\n92\n46\n679\n165\n146\n149\n93\n2211\n6495\n307\n139\n253\n123\n21\n84\n98\n13\n230\n\
+       277\n1\n986\n414\n144\n210\n0\n172\n165\n25\n97\n335\n558\n350\n369\n511\n245\n345\n8\n139\n55\n318\n2614\n\
+       1792\n16\n345\n123\n176\n1171\n20\n199\n147\n115\n335\n23\n847\n94\n315\n391\n176\n282\n459\n197\n978\n281\n\
+       27\n26\n281\n8124\n3\n430\n510\n61\n67\n17\n1601\n362\n202\n50\n10\n346\n258\n664\nNULL\n22\n164\n448\n365\n\
+       1640\n671\n203\n2087\n10060\n1015\n913\n9840\n16\n496\n264\n38\n1";
+    assert_eq!(expected, actual);
+    Ok(())
+}
```

Comment:
What do you think if we run against a controlled dataset instead of `aggregate_test_100`? I can't tell whether `expected` is correct. Wouldn't it make it easier if we use something like the `query_not` test is doing, on which it creates a temporary table and runs against that?

Reply:
Difficult to tell what it is happening here.⬆️

Reply:
Ok, let me see if I can improve that test or upload new data (like a subset of the original)

Reply:
it is more important to test the different cases here than have a large dataset. You are already providing excellent coverage with the other tests, this is more like an end-to-end test to verify the result in integration, so we do not need to cover many situations.

Reply:
Cleaned up.   :)